native-dnn/src/main/java/com/intel/analytics/bigdl/mkl/hardware/platform/linux/LinuxAffinity.java

```diff
@@ -2,84 +2,185 @@
 
 import com.intel.analytics.bigdl.mkl.hardware.platform.IAAffinity;
 
-import java.util.Arrays;
-import java.util.BitSet;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
 
 public class LinuxAffinity implements IAAffinity {
-    public static native int setAffinity0(int[] sets);
-    public static native int getAffinity0(int[] sets, int cpuCounts);
-    public static native void setOmpAffinity0(int[] sets);
+    public static native int setAffinity0(int[] set);
+    public static native int getAffinity0(int[] set);
+    public static native int[] setOmpAffinity0(int[] set);
+    public static native int[] getOmpAffinity0(int cpuCounts);
 
     private int cpuCounts;
-    private boolean[] availableCPUs;
+    private int[] cpuList;
+    private ConcurrentHashMap<Integer, List<Long>> availableCores;
 
     public LinuxAffinity(int cpuCounts) {
         this.cpuCounts = cpuCounts;
-        this.availableCPUs = new boolean[cpuCounts];
-        Arrays.fill(this.availableCPUs, false);
+
+        this.cpuList = getAffinity();
+
+        this.availableCores = new ConcurrentHashMap<Integer, List<Long>>();
```

Comment:
what's the difference between cpu and core in your code context ?

Reply:
The `cpuList` term comes from affinity. I have changed them to cores here.